tests/components/vesync/common.py

```diff
@@ -1,7 +1,28 @@
 """Common methods used across tests for VeSync."""
 import json
 
-from tests.common import load_fixture
+import requests_mock
+
+from tests.common import load_fixture, load_json_object_fixture
+
+ALL_DEVICES = load_json_object_fixture("vesync/vesync_api_call__devices.json")
+ALL_DEVICE_NAMES: list[str] = [
+    dev["deviceName"] for dev in ALL_DEVICES["result"]["list"]
+]
+
+
+def mock_devices_response(
+    requests_mock: requests_mock.Mocker, device_name: str
+) -> None:
+    """Build a response for the Helpers.call_api method."""
+    device_list = []
+    for device in ALL_DEVICES["result"]["list"]:
+        if device["deviceName"] == device_name:
+            device_list.append(device)
+    requests_mock.post(
```

Comment:
>     requests_mock.post(
>         "https://smartapi.vesync.com/cloud/v1/deviceManaged/devices",
>         json={"code": 0, "result": {"list": device_list}},
>     ) 

Excuse my ignorance about "tests" which I'm trying to learn how to implement them. But one thing called my attention here. 

Are these lines testing the cloud API directly?  

Because I thought Homeassistant was not allowed to do that. no IOs. All interaction should be done with the libraries only (pyvesync).

So, in my head, the tests should be testing the library objects, not the cloud api (which is already tested while developing the library)

Does my thoughts make sense in this case? please help me understand.
I know how to write some code, but creating "tests" for it , I'm trying to learn about it.

Reply:
These lines do not test the cloud API.
They "inject" data so that `pyvesync` receives a reply without any IO.

It would indeed be better to avoid this altogether but with some library it can be very tricky.